### PR TITLE
Add testing framework and some initial tests

### PR DESCRIPTION
**Pull request by bioball**
This adds a testing framework for the LSP, and also adds some initial tests.

This is built on top of https://github.com/apple/pkl-lsp/pull/5, so skip the first commit when reviewing this one.

### CODE REVIEW

File: src/test/kotlin/org/pkl/lsp/FakeLanguageClient.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+package org.pkl.lsp
+
+import java.util.concurrent.CompletableFuture
+import org.eclipse.lsp4j.MessageActionItem
+import org.eclipse.lsp4j.MessageParams
+import org.eclipse.lsp4j.PublishDiagnosticsParams
+import org.eclipse.lsp4j.ShowMessageRequestParams
+import org.eclipse.lsp4j.services.LanguageClient
+
+object FakeLanguageClient : LanguageClient {
```
Comment (stackoverflow) [inline src/test/kotlin/org/pkl/lsp/FakeLanguageClient.kt:25]: nit: `TestLanguageClient` is a more descriptive name.

File: src/test/kotlin/org/pkl/lsp/GoToDefinitionTest.kt [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        name<caret> = "Bob"
+      }
+    """
+        .trimIndent()
+    )
+    val resolved = goToDefinition().single()
+    assertThat(resolved).isInstanceOf(PklClassProperty::class.java)
+    resolved as PklClassProperty
+    assertThat(resolved.name).isEqualTo("name")
+    assertThat(resolved.parentOfTypes(PklClass::class)!!.name).isEqualTo("MyClass")
```
Comment (stackoverflow) [inline src/test/kotlin/org/pkl/lsp/GoToDefinitionTest.kt:57]: nit: you can use `parentOfType<PklClass>()` here.

File: src/test/kotlin/org/pkl/lsp/GoToDefinitionTest.kt [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+      """
+      function foo(myParam: String) =
+        myParam<caret> + 5
+    """
+        .trimIndent()
+    )
+    val resolved = goToDefinition().single()
+    assertThat(resolved).isInstanceOf(PklTypedIdentifier::class.java)
+    resolved as PklTypedIdentifier
+    assertThat(resolved.identifier!!.text).isEqualTo("myParam")
```
Comment (stackoverflow) [inline src/test/kotlin/org/pkl/lsp/GoToDefinitionTest.kt:88]: nit: we could also assert that `resolved.typeAnnotation!!.type!!.render() == "String"`

**Review by stackoverflow — APPROVED** [review]
Looks good! Left some nits.

File: src/test/kotlin/org/pkl/lsp/GoToDefinitionTest.kt (hunk 2 above)
Comment (bioball) [reply src/test/kotlin/org/pkl/lsp/GoToDefinitionTest.kt:57]: This wasn't working for me (I was getting a null pointer exception).

Gonna merge this as-is for now.

**Pull request merged**